src/MIDebugEngine/Engine.Impl/DebuggedProcess.cs

```diff
@@ -1422,29 +1422,53 @@ public ReadOnlyCollection<RegisterGroup> GetRegisterGroups()
             await _breakpointManager.EnableAfterFuncEvalAsync();
         }
 
+        /// <summary>
+        /// Finds the line associated with a start address.
+        /// </summary>
+        public async Task<uint> LineForStartAddress(string file, ulong startAddress)
+        {
+            List<ulong> addresses = new List<ulong>();
+            SourceLineMap srcLines = await SourceLineCache.GetLinesForFile(file);
+            if (srcLines == null || srcLines.Count == 0)
+            {
+                srcLines = await SourceLineCache.GetLinesForFile(System.IO.Path.GetFileName(file));
+            }
+            if (srcLines == null || srcLines.Count == 0)
+            {
+                return 0;
+            }
+
+            SourceLine srcLine;
+            if (srcLines.TryGetValue(startAddress, out srcLine))
+            {
+                return srcLine.Line;
+            }
+            return 0;
+        }
+
         public async Task<List<ulong>> StartAddressesForLine(string file, uint line)
         {
             List<ulong> addresses = new List<ulong>();
             var srcLines = await SourceLineCache.GetLinesForFile(file);
```

Comment:
nit: code is easier to read if it uses the real type (SourceLineMap in this case) instead of 'var' in cases where the type isn't obvious. Especially since it is used both here and on line 1460.